gateway/channel_users.go

```diff
@@ -0,0 +1,126 @@
+package gateway
+
+import "github.com/nlopes/slack"
+
+// getChannelUsersFromAPI queries the Slack API for a list of users in the given channel, returning
+// SlackUser objects for each one
+func (sc *SlackClient) getChannelUsersFromAPI(channelID string) (users []*SlackUser, err error) {
+	hasMore := true
+	guicp := &slack.GetUsersInConversationParameters{
+		ChannelID: channelID,
+		Limit:     1000,
+	}
+	for hasMore {
+		var userIDs []string
+		userIDs, guicp.Cursor, err = sc.client.GetUsersInConversation(guicp)
+		if err != nil {
+			return
```

Comment:
If we have an error,  do we want to return users that we've already resolved? Say an error we encounter in the second page.

Reply:
it's probably fine? if `err != nil` from this function, I think any callers (well the only caller is below) will discard the first return value anyway